### PR TITLE
Replaced deprecated nccl variables

### PR DESCRIPTION
**Pull request by Sergei-Lebedev**
PR reflects changes in nccl 2.5 where NCCL_LL_THRESHOLD and NCCL_TREE_THRESHOLD were replaced with NCCL_PROTO and NCCL_ALGO

### CODE REVIEW

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI status: <b>UNKNOWN</b> (click for details)</summary>

**Note:** the artefacts will be deleted after 03-Dec-2019

Artefact file | Status
-- | --
[build_nccl.log](https://gist.githubusercontent.com/mellanox-github/e1624cd6ed880e2a697561c16c566eb5/raw/52d79aa32b44090f49e341d5d3a3433754c4d3df/build_nccl.log) | :heavy_check_mark: PASS
[build_nccl_rdma_sharp_plugins.log](https://gist.githubusercontent.com/mellanox-github/e1624cd6ed880e2a697561c16c566eb5/raw/e92d2b2957f3e226857a3da2fd4ce796c1c2c786/build_nccl_rdma_sharp_plugins.log) | :heavy_check_mark: PASS
[configure_sharp_startup.log](https://gist.githubusercontent.com/mellanox-github/e1624cd6ed880e2a697561c16c566eb5/raw/761210eef7b1a70e239c3ea24609a7888443616c/configure_sharp_startup.log) | :heavy_check_mark: PASS
[run_nccl_test.log](https://gist.githubusercontent.com/mellanox-github/e1624cd6ed880e2a697561c16c566eb5/raw/052855f8f918426a7233a2715fcc9cac372880fe/run_nccl_test.log) | :question: UNKNOWN

**Comment by mellanox-github** [comment]
Mellanox CI FAILed.

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI status: <b>PASS</b#  > (click for details)</summary>

**Note:** the artefacts will be deleted after 03-Dec-2019

Artefact file | Status
-- | --
[build_nccl.log](https://gist.githubusercontent.com/mellanox-github/6c6fe80b5b6f70436a451710863043eb/raw/37f5ab322e7222453af980fa795c3c52a1ce2fe9/build_nccl.log) | :heavy_check_mark: PASS
[build_nccl_rdma_sharp_plugins.log](https://gist.githubusercontent.com/mellanox-github/6c6fe80b5b6f70436a451710863043eb/raw/8aaf806fde855637057a60cdaa642377e2f4f315/build_nccl_rdma_sharp_plugins.log) | :heavy_check_mark: PASS
[configure_sharp_startup.log](https://gist.githubusercontent.com/mellanox-github/6c6fe80b5b6f70436a451710863043eb/raw/855c555391e9d07fc88e13cc77b740d24d1cadac/configure_sharp_startup.log) | :heavy_check_mark: PASS
[configure_sharp_stop.log](https://gist.githubusercontent.com/mellanox-github/6c6fe80b5b6f70436a451710863043eb/raw/53cf6c4c658025370cad5c51484bbe84aaa9ba09/configure_sharp_stop.log) | :heavy_check_mark: PASS
[run_nccl_test.log](https://gist.githubusercontent.com/mellanox-github/6c6fe80b5b6f70436a451710863043eb/raw/429e1e53df3930a29cb5825d2882bf5b12f826c5/run_nccl_test.log) | :heavy_check_mark: PASS

**Comment by mellanox-github** [comment]
Mellanox CI PASSed.

**Comment by Sergei-Lebedev** [comment]
@itemko @bureddy pls review

File: .ci/run_nccl_test.sh [hunk 1]
```diff
@@ -127,8 +127,9 @@ export LD_LIBRARY_PATH="${NCCL_DIR}/lib:${NCCL_RDMA_SHARP_PLUGINS_DIR}/lib:${LD_
     echo_hash_line
 
     MPIRUN_OPTIONS_SPECIFIC="\
-    -x NCCL_LL_THRESHOLD=0 \
-    -x NCCL_TREE_THRESHOLD=1000000000 \
+    -x NCCL_PROTO=Simple \
+    -x NCCL_ALGO=CollNet \
```
Comment (itemko) [inline .ci/run_nccl_test.sh]: `NCCL_ALGO=CollNet` is it not documented yet? Don't see such value in the official documentation https://docs.nvidia.com/deeplearning/sdk/nccl-developer-guide/docs/env.html#nccl-algo

File: .ci/run_nccl_test.sh [hunk 2]
```diff
@@ -160,8 +161,9 @@ fi
     echo_hash_line
 
     MPIRUN_OPTIONS_SPECIFIC="\
-    -x NCCL_LL_THRESHOLD=1000000000 \
-    -x NCCL_TREE_THRESHOLD=1000000000 \
+    -x NCCL_PROTO=LL \
```
Comment (itemko) [inline .ci/run_nccl_test.sh]: Do we need test with LL128 or comma-separated list?

File: .ci/run_nccl_test.sh [hunk 3]
```diff
@@ -160,8 +161,9 @@ fi
...
+    -x NCCL_ALGO=CollNet \
+    -x NCCL_COLLNET_ENABLE=1 \
```
Comment (itemko) [inline .ci/run_nccl_test.sh]: As far as I see it is not yet documented - what's the difference between  `NCCL_ALGO=CollNet + NCCL_COLLNET_ENABLE=1` and `NCCL_ALGO=CollNet`?

File: .ci/run_nccl_test.sh (hunk 1 above)
Comment (Sergei-Lebedev) [reply .ci/run_nccl_test.sh]: Yes, it's not documented cause nccl 2.5 with collnet support has not been released yet.

File: .ci/run_nccl_test.sh (hunk 2 above)
Comment (Sergei-Lebedev) [reply .ci/run_nccl_test.sh]: CollNet algorithm in NCCL doesn't support LL128 protocol, so I'm not sure it's important at the moment. We can add test with multiple protocols/algos but we also can add some other tests. Maybe it's better to make a separate PR for the new tests. What do you think?

File: .ci/run_nccl_test.sh (hunk 3 above)
Comment (Sergei-Lebedev) [reply .ci/run_nccl_test.sh]: Just NCCL_COLLNET_ENABLE=1 enables collective plugins support in NCCL, but NCCL can choose any available algorithm. By setting both flags we enable collective network support (NCCL_COLLNET_ENABLE=1 )and also force nccl to use CollNet for any message size (NCCL_ALGO=CollNet).

File: .ci/run_nccl_test.sh (hunk 2 above)
Comment (itemko) [reply .ci/run_nccl_test.sh]: Separate PR for new tests will be OK - I've submitted https://trello.com/c/W7IudX5a

**Comment by itemko** [comment]
bot:mlx:test

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI status: <b>PASS</b#  > (click for details)</summary>

**Note:** the artefacts will be deleted after 04-Dec-2019

Artefact file | Status
-- | --
[build_nccl.log](https://gist.githubusercontent.com/mellanox-github/026c394df6cddcba0e1cfdbe8305355f/raw/a772edf064d5cf460d89eefd5daebf11ba6b3a51/build_nccl.log) | :heavy_check_mark: PASS
[build_nccl_rdma_sharp_plugins.log](https://gist.githubusercontent.com/mellanox-github/026c394df6cddcba0e1cfdbe8305355f/raw/a71a9c77d0df2c1a9a7eb9df9b3f1fe12dc7dc50/build_nccl_rdma_sharp_plugins.log) | :heavy_check_mark: PASS
[configure_sharp_startup.log](https://gist.githubusercontent.com/mellanox-github/026c394df6cddcba0e1cfdbe8305355f/raw/f6c027d60bb8c5ee7e6f519fab5604973882adb3/configure_sharp_startup.log) | :heavy_check_mark: PASS
[configure_sharp_stop.log](https://gist.githubusercontent.com/mellanox-github/026c394df6cddcba0e1cfdbe8305355f/raw/50236eba921ab0ac0b4a239bc7630b246e3c5970/configure_sharp_stop.log) | :heavy_check_mark: PASS
[run_nccl_test.log](https://gist.githubusercontent.com/mellanox-github/026c394df6cddcba0e1cfdbe8305355f/raw/7e9045f00024de4051cbd8439d85e4a2faa3bb1d/run_nccl_test.log) | :heavy_check_mark: PASS

**Comment by mellanox-github** [comment]
Mellanox CI PASSed.

**Pull request merged**